tests/rptest/services/kafka_cli_consumer.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2020 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import re
+import threading
+from ducktape.services.background_thread import BackgroundThreadService
+
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+from ducktape.utils.util import wait_until
+
+
+class KafkaCliConsumer(BackgroundThreadService):
+    def __init__(self,
+                 context,
+                 redpanda,
+                 topic,
+                 group=None,
+                 partitions=[],
+                 consumer_properties={},
+                 isolation_level=None):
+        super(KafkaCliConsumer, self).__init__(context, num_nodes=1)
+        self._redpanda = redpanda
+        self._topic = topic
+        self._stopping = threading.Event()
+        self._done = False
+        self._offset = dict()
+        self._group = group
+        self._isolation_level = isolation_level
+        self._consumer_properties = consumer_properties
+        self._partitions = partitions
+
+        self._pid = None
+        self._cli = KafkaCliTools(self._redpanda)
+        self._messages = []
+        assert self._partitions is not None or self._group is not None, "either partitions or group have to be set"
```

Comment:
_partitions is never used in this class.
Also, _partitions can only be None if the user explicitly passes partitions=None. So is _partitions needed in the assert?

Reply:
good point, i missed that